internal/apptoken/app_token_usage.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package apptoken
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/apptoken/store"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const appTokenUsageTableName = "app_token_usage"
+
+// AppTokenUsage defines an app token usage entity for storage.
+type AppTokenUsage struct {
+	*store.AppTokenUsage
+	tableName string
+}
+
+// NewAppToken creates an in-memory app token with options.  Supported options:
+// WithName, WithDescription
+func NewAppTokenUsage(ctx context.Context, appTokenId, clientTcpAddress, requestMethod, requestPath string, createdTime time.Time, _ ...Option) (*AppTokenUsage, error) {
+	const op = "apptoken.NewAppToken"
+	switch {
+	case appTokenId == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing app token id")
+	case clientTcpAddress == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing cleint tcp address")
+	case requestMethod == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing request method")
+	case requestPath == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing request path")
+	case createdTime.IsZero():
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing created time")
+	}
+
```

Comment:
should we validate the the clientTcpAddress via `net.ParseIP(...)`

Reply:
Makes sense. I added a validation